base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -1436,96 +1437,107 @@ impl BlockchainBackend for LMDBDatabase {
         start: u64,
         end: u64,
         deleted: &Bitmap,
-    ) -> Result<(Vec<PrunedOutput>, Vec<Bitmap>), ChainStorageError>
+    ) -> Result<(Vec<PrunedOutput>, Bitmap), ChainStorageError>
     {
         let txn = ReadTransaction::new(&*self.env)?;
-        if let Some(start_height) = lmdb_first_after(&txn, &self.output_mmr_size_index, &(start + 1).to_be_bytes())? {
-            let end_height: u64 =
-                lmdb_first_after(&txn, &self.output_mmr_size_index, &(end + 1).to_be_bytes())?.unwrap_or(start_height);
-
-            let previous_mmr_count = if start_height == 0 {
-                0
-            } else {
-                let header: BlockHeader =
-                    lmdb_get(&txn, &self.headers_db, &(start_height - 1))?.expect("Header should exist");
-                debug!(target: LOG_TARGET, "Previous header:{}", header);
-                header.output_mmr_size
-            };
+        let start_height = lmdb_first_after(&txn, &self.output_mmr_size_index, &(start + 1).to_be_bytes())?
+            .ok_or_else(|| {
+                ChainStorageError::InvalidQuery(format!(
```

Comment:
Is there not error to handle this conversion?
I see prev it just had a "?"

Reply:
No. If it was None it would return `  Ok((vec![], vec![]))`